bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -40,10 +41,15 @@
     public static final int MIN_ELEVATION_WARN_LIMIT = 42;
 
     public void propertyChange(PropertyChangeEvent evt) {
-        Variant v = (Variant) evt.getSource();
-        MarkerManager mm = v.getSchedule().getMarkerManager();
+        final Variant  v               = (Variant) evt.getSource();
+        final Schedule schedule        = v.getSchedule();
+        final MarkerManager mm         = schedule.getMarkerManager();
         mm.clearMarkers(this, v);
 
+        final TwilightBoundedNight tbn = Twilight.forTime(schedule.getStart(), schedule.getSite());
```

Comment:
Moving this line up out of the big loop over allocations and using `schedule.getStart()` is the important bit of this PR.

Reply:
It is now consistent with the [cache entry calculations](https://github.com/gemini-hlsw/ocs/blob/develop/bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/Variant.java#L774).